app/invocation/invocation_action_card.tsx

```diff
@@ -319,122 +319,161 @@ export default class InvocationActionCardComponent extends React.Component<Props
+          ? next.timestamp
+          : event.timestamp;
+
+      return event;
+    });
+
+    const allUnixTimestamp = filteredEvents.map((e) => timestampToUnixSeconds(e.timestamp));
+    const startUnixTimestamp = Math.min(...allUnixTimestamp);
+    const endUnixTimestamp = Math.max(...allUnixTimestamp);
+    const totalDuration = Math.max(0, endUnixTimestamp - startUnixTimestamp);
 
     let offset = 0;
     return (
-      <div>
-        <div className="metadata-detail">
-          <span className="label">
-            Total
-            {startTimestamp && <> @ {format.formatTimestamp(startTimestamp)}</>}
-          </span>
-          <span className="bar-description">{format.durationSec(totalDuration)} (100%)</span>
-        </div>
-        <div className="action-timeline">
-          <div
-            className="timeline-event"
-            title={`Total: (${format.durationSec(totalDuration)}, 100%)`}
-            style={{ flex: `1 0 0`, backgroundColor: "green" }}></div>
-        </div>
-        {filteredEvents.map((event, i) => {
-          // Don't render the end marker.
-          if (i == filteredEvents.length - 1) return null;
-
-          const next = filteredEvents[i + 1];
-          const duration = durationSeconds(event.timestamp!, next.timestamp!);
-          const weight = duration / totalDuration;
-          offset += weight;
-          return (
-            <div>
-              <div className="metadata-detail">
-                <span className="label">
-                  {event.name}
-                  {event.timestamp && <> @ {format.formatTimestamp(event.timestamp)}</>}
-                </span>
-                <span className="bar-description">
-                  {format.compactDurationSec(duration)} ({(weight * 100).toFixed(1)}%)
-                </span>
-              </div>
-              <div className="action-timeline">
-                <div
-                  className="timeline-event-gray"
-                  title={`${event.name} (${format.durationSec(duration)}, ${(weight * 100).toFixed(2)}%)`}
-                  style={{ flex: `${offset - weight} 0 0`, backgroundColor: `rgba(0, 0, 0, .1)` }}></div>
-                <div
-                  className="timeline-event"
-                  title={`${event.name} (${format.durationSec(duration)}, ${(weight * 100).toFixed(2)}%)`}
-                  style={{ flex: `${weight} 0 0`, backgroundColor: event.color }}></div>
-                <div
-                  className="timeline-event-gray"
-                  title={`${event.name} (${format.durationSec(duration)}, ${(weight * 100).toFixed(2)}%)`}
-                  style={{ flex: `${1 - offset} 0 0`, backgroundColor: `rgba(0, 0, 0, .1)` }}></div>
+      <>
+        <div className="metadata-title">Timeline</div>
+        <div>
+          <div className="metadata-detail">
+            <span className="label">Total @ {format.formatTimestamp(filteredEvents[0].timestamp)}</span>
```

Comment:
nit: The `Total @ <timestamp>` feels a bit awkward, maybe just make the label empty. We show the start timestamp for the queued event so I think this is redundant anyway

```suggestion
            <span className="label"></span>
```

Reply:
I will remove the timestamp for now. It's a bit weird to just see a duration on the right column without a label on the left.

Reply:
Ah yeah sorry, that's what I meant :D